packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -339,10 +386,44 @@ export class RichTextEditor extends RichText implements ErrorPattern {
         );
     }
 
+    public mentionChange(e: CustomEvent<MentionDetail>): void {
+        this.mentionPropCommand.command({
+            href: e.detail.href,
+            label: e.detail.displayName
+        });
+        this.openMentionPopup = false;
+    }
+
     protected override updateView(): void {
         this.setMarkdown(this.getMarkdown());
     }
 
+    protected override updateMentionConfig(): void {
+        super.updateMentionConfig();
+        this.mentionElements.forEach(element => {
+            this.mentionInternalsMap.clear();
+            if (
+                element.mentionInternals.pattern
+                && element.mentionInternals.mentionConfig
+            ) {
+                this.mentionInternalsMap.set(
+                    element.mentionInternals.character,
+                    element.mentionInternals
+                );
+            }
+            this.activeConfiguration = this.activeChar
+                ? this.mentionInternalsMap.get(this.activeChar)?.mentionConfig
+                    ?.mappingConfigs
+                : undefined;
```

Comment:
I think this mentionsInternalsMap stuff is a symptom of Mapping Extension not being configured correctly as discussed here: https://github.com/ni/nimble/pull/1683#discussion_r1410031640

As described in the comment, similar to the Markdown Parser, the TipTap Extension should not be dealing with mentionInternals, etc.

Reply:
As we have refactored the entire logic, I am closing this thread